ports/stm32/boards/LEGO_HUB_NO7/README.md

```diff
@@ -0,0 +1,135 @@
+LEGO Hub No.7
+=============
+
+This board definition is for the LEGO Hub No. 7, a LEGO control unit with 1 button,
+2 RGB LEDs, 2 Powered Up ports, 6-DOF sensor, Bluetooth, USB, 4MiB external SPI
+flash storage, and a rechargeable battery.
+
+Features that are currently supported:
+- standard MicroPython
+- machine and bluetooth modules
+- filesystem
+- USB VCP, MSC and HID
+
+The Hub has a bootloader preinstalled at 0x08000000 (which is 32kiB in size) which
+cannot be erased.  This bootloader is entered by holding down the button for 5 seconds,
+at which point the USB DFU device appears.  If the battery is installed then the
+RGB LED will flash purple. If the battery is not installed, the LED will flash orange
+briefly and then the hub will turn off (so having the battery installed is required).
+When this bootloader is active, the flash from 0x08008000 and up can be erased
+and programmed via USB DFU.
+
+The built-in bootloader has some drawbacks: it cannot be entered programmatically,
+and it does not keep the Hub powered up when running from battery (which requires
+keeping BAT_PWR_EN high).  As such, this board is configured to work with mboot as
+a secondary bootloader: mboot is placed at 0x08008000 and the main application
+firmware at 0x08010000.  When mboot is installed it can be entered programatically
+via machine.bootloader().
+
+Backing up original Hub firmware
+--------------------------------
+
+Before installing MicroPython it is advised to backup the original LEGO firmware that
+the Hub comes installed with.  To do this, enter the built-in bootloader by holding
+down the Bluetooth button for 5 seconds while powering up the Hub via USB (you may
+need to take out the battery and disconnect USB to power off the Hub first).  Then
```

Comment:
I think these instructions to enter the built-in bootloader are different on HUB_NO7.  There is no Bluetooth button.